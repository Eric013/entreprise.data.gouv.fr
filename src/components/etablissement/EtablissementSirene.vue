<template>
  <div class="company">
    <div class="company-container panel">
      <etablissement-sirene-contact />
      <etablissement-sirene-info />
    </div>
  </div>
</template>

<script>
import Filters from '@/components/mixins/filters.js'
import EtablissementSireneContact from '@/components/etablissement/etablissementSirene/EtablissementSireneContact'
import EtablissementSireneInfo from '@/components/etablissement/etablissementSirene/EtablissementSireneInfo'

export default {
  name: 'EtablissementSirene',
  components: {
    'EtablissementSireneContact': EtablissementSireneContact,
    'EtablissementSireneInfo': EtablissementSireneInfo
  }
}
</script>

<style lang="scss" scoped>
  .panel + .panel {
    margin-left: 2em;
  }

  @media (max-width: $tablet) {
    .panel + .panel {
      margin-left: 0;
    }
  }
</style>
