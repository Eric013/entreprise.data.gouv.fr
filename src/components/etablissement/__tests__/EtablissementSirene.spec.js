import { createLocalVue, shallow } from '@vue/test-utils'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import Vuex from 'vuex'
import EtablissementSirene from '@/components/etablissement/EtablissementSirene.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')
jest.mock('mapbox-gl', () => ({
  supported: jest.fn(),
  Popup: jest.fn()
}))
describe('Etablissement.vue', () => {
  let storeMocks
  let wrapperEtablissement
  let etablissement

  beforeEach(() => {
    storeMocks = createStoreMocks()

    wrapperEtablissement = shallow(EtablissementSirene, {
      localVue,
      store: storeMocks.store
    })
    etablissement = wrapperEtablissement.vm
  })

  test('Computed value Result returns the right store getter', () => {
    expect(etablissement.result).toBe(storeMocks.store.getters.singlePageResultEtablissement)
  })

  test('Computed value coordinates returns the coordinates if both longitude and latitude exists', () => {
    etablissement.result.longitude = -30
    etablissement.result.latitude = -42
    wrapperEtablissement.update()
    expect(etablissement.coordinates).toEqual([-30, -42])

    etablissement.result.longitude = -30
    etablissement.result.latitude = null
    wrapperEtablissement.update()
    expect(etablissement.coordinates).toBeNull()

    etablissement.result.longitude = null
    etablissement.result.latitude = -42
    wrapperEtablissement.update()
    expect(etablissement.coordinates).toBeNull()
  })
})

describe('EtablissementSirene.vue Snapshot testing', () => {
  const storeMocks = createStoreMocks()

  const wrapperES = shallow(EtablissementSirene, {
    localVue,
    store: storeMocks.store
  })

  test('It match the snapshot', () => {
    expect(wrapperES.vm.$el).toMatchSnapshot()
  })
})