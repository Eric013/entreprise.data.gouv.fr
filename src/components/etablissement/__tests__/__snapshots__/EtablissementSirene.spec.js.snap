// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementSirene.vue Snapshot testing It match the snapshot 1`] = `
<div
  class="company"
>
  <section
    class="section-grey"
  >
    <div
      class="container company-container"
    >
      <!---->
       
      <!---->
    </div>
     
    <div
      class="container company-container company-container__extra"
    >
      <!---->
    </div>
     
    <div
      class="container company-container company-container__map"
    >
      <!---->
    </div>
  </section>
</div>
`;
